cli/command/image/save.go

```diff
@@ -41,6 +44,14 @@ func runSave(dockerCli command.Cli, opts saveOptions) error {
 		return errors.New("cowardly refusing to save to a terminal. Use the -o flag or redirect")
 	}
 
+	dir := filepath.Dir(opts.output)
+	if dir != "" && dir != "." {
+		_, err := os.Stat(dir)
+		if err != nil && os.IsNotExist(err) {
+			return fmt.Errorf("unable to save to %s: directory does not exist", opts.output)
+		}
+	}
```

Comment:
I think this should be a separate function  `func validateOutputPath(...) error`.